src/context/apiClient/apiClientContextController/interceptors/responseInterceptors.ts

```diff
@@ -1,5 +1,6 @@
 import { AxiosResponse } from 'axios';
 
-export const responseSuccessInterceptor = (response: AxiosResponse) => response;
+export const responseSuccessInterceptor = (response: AxiosResponse) =>
+  response.headers['content-type'] === 'application/json' ? response : Promise.reject(response);
```

Comment:
what's the reason for that change?

Reply:
Yeah, so MSW works only for development env. Then none of requests works on built app, then axios return HTML content instead of json. Example: useUsers() in Home.tsx, return HTML response (index.html with loader) and whole app crashes (because it should return GetUsersResponse type of undefined). This is my workout around, adding 'application/json' to request headers didn't work.

issue: https://github.com/axios/axios/issues/889

Reply:
Ok, i checked this again and i just added new commit, i resolved problem in other way. I reverted above changes and added changes ([COMMIT](https://github.com/TheSoftwareHouse/react-starter-boilerplate/pull/40/commits/bf7d5e3002606f2a4e732029be89f3f9f7f638e4)):

Changed **REACT_APP_API_URL** env from `mock` to `http://mock`. Thanks that, all axios requests won't hit URL like `localhost:3000/mock/users/?page=0&count=5` (it's relative path, it tries to hit localhost, which actually serve HTML (not json endpoint), and FYI: it's not covered by MSW because it's production build), but will hit url like `http://mock/users/?page=0&count=5` (absolute path) which doesn't exist and it fails, thanks that returning undefined to `data` var.

Reply:
nice one!